components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -51,7 +286,72 @@ func (ss *BatchCheckService) RunBastionCheck(check string, config models.Config,
 		resp[i].CheckType = check
 	}
 	resultChan <- resp
+}
+
+func (ss *BatchCheckService) GetPortsToOpenForCheck(check string) map[string]map[string][]int {
+	return ss.getCheckInstance(check).GetPortsForMockServer()
+}
+
+func (ss *BatchCheckService) getDeploymentState(config models.Config) (string, error) {
+	if config.Hardware.AutomateNodeCount > 0 {
+		for index, ip := range config.Hardware.AutomateNodeIps {
+			url := fmt.Sprintf("http://%s:%s%s", ip, ss.port, constants.STATUS_API_PATH)
+			resp, err := ss.httpRequestClient.MakeRequest(http.MethodGet, url, nil)
+			if err != nil && index < len(config.Hardware.AutomateNodeIps)-1 {
+				ss.log.Error("Error while calling status api from batch check service:", err)
+				continue
+			}
+			if err != nil && index == len(config.Hardware.AutomateNodeIps)-1 {
+				errMsg := "received no response for status api from any automate nodes"
+				ss.log.Error(errMsg)
+				return "", errors.New(errMsg)
+			}
+			var statusApiResponse models.StatusApiResponse
+			err = json.NewDecoder(resp.Body).Decode(&statusApiResponse)
+			if err != nil {
+				ss.log.Error("Error while reading unmarshalling response of status api from batch check service:", err)
+				return "", err
+			}
```

Comment:
create this portion as function like: getStatusFromNode(ip) -> result

Reply:
Done